authors/apps/authentication/views.py

```diff
@@ -73,3 +89,74 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+@api_view(['GET', 'POST'])
+def password_reset(request):
+    if request.method == 'POST':
+ 
+        token = uuid.uuid4()
+        email = request.data.get('email')
+
+        user = get_object_or_404(User, email=email)
+        
+        Token.objects.create(
+            token=token,
+            email=user.email
+        )
+
+        url = f'http://127.0.0.1:8000/api/reset_password/?token={token}'
+        
+        message = f'Please click the link to reset your password\n {url} \n {token}'
+
+        send_mail(
+            'Password Reset Link',
+            message,
+            settings.EMAIL_HOST_USER,
+            [user.email],
+            fail_silently=False,
+        )
+
+        return Response({"message": "email sent", "data": request.data})
+    return Response({"message": "Great Work, TeamOdin"})
+
+@api_view(['GET'])
+def reset_password(request):
+
+    token = request.GET['token'].strip()
+    token = get_object_or_404(Token, token=token)
+
+    time_created = token.created_at
+    current_time = now()
+    diff = current_time - time_created
+    hours = round(diff.total_seconds()/3600)
+    
+    data = {
+                'token_status': False
+        }
+
+    if token:
+        data['token_status'] = True
+        if hours > 4: 
+            return Response({'message': 'Password reset token only valid for 4 hours, please re-try to reset your password', "data": data})
+        
+        return Response({"message": "Token Exists And It's Valid, allow password reset", "data": data})
+    else:
+        return Response({"message": "Token  does not Exist, password reset request denied", "data": data})
+
+@api_view(['POST'])
+def change_passowrd(request): 
+    if request.method == 'POST':
+        data = {}
+        user = User.objects.get(email=request.POST['email'])
+        if user: 
+            password = request.POST['password'].strip()
+            if validate_password(password) == True:
+                user.set_password(password)
```

Comment:
Serializers would be a better place for doing this sort of validation, instead of doing it here in the views.